### DISCUSSION TITLE
How many pulse counter units are used/available for other purposes?

### DISCUSSION THREAD

**Discussion by wiegleyj** [gin66/FastAccelStepper#167] [General]
I see that this library connects the PWM output to the pulse counter(s). I also need pulse counters for the other portions of my application. I need only one stepper motor though. The ESP32-S3 has four pulse counter unit. If I setup/use a single stepper motor with this library how many of the four pulse counter units are still available for other uses?

**Comment by gin66** [comment]
FastAccelStepper allocates one pulse counter per stepper starting with 0. This means pulse counters 1 to 3 are available.

More info is in the [readme](https://github.com/gin66/FastAccelStepper/tree/master#esp32-1)
